core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -793,6 +793,7 @@ class ReplicaManagerTest {
 
     try {
       val brokerList = Seq[Integer](0, 1).asJava
+      when(replicaManager.metadataCache.getAliveBrokerEpoch(1)).thenReturn(Some(brokerEpoch))
```

Comment:
Why is this change needed?

Reply:
Same as above. In `ReplicaManagerTest#setupReplicaManagerWithMockedPurgatories`, it uses `mock(classOf[MetadataCache])`.

https://github.com/apache/kafka/blob/94a1bfb1281f06263976b1ba8bba8c5ac5d7f2ce/core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala#L3141

Reply:
Could you please move the mock to line#3154? We should mock the method after creating the mocked `metadataCache`